src/ui/navigator/navigationBar/VPNBottomNavigationBar.qml

```diff
@@ -150,26 +150,18 @@ Rectangle {
         id: navBarButtonGroup
     }
 
-    VPNFilterProxyModel {
-        id: messagesModel
-        source: VPNAddonManager
-        filterCallback: obj => { return obj.addon.type === "message" && !obj.addon.isRead }
-        Component.onCompleted: {
-            messagesNavButton._hasNotification = Qt.binding(() => { return messagesModel.count > 0} )
-        }
-    }
-
     Connections {
         target: VPNSettings
-        function onReadAddonMessagesChanged() {
-            root.getUnreadNotificationStatus()
-        }
-        function onDismissedAddonMessagesChanged() {
+        function onAddonSettingsChanged() {
             root.getUnreadNotificationStatus()
         }
     }
 
     function getUnreadNotificationStatus() {
         messagesNavButton._hasNotification = VPNAddonManager.reduce((addon, initialValue) => initialValue + (addon.type === "message" && !addon.isRead), 0)
```

Comment:
I know this is not part of this PR, but it's confusing, 1. Can we change `initialValue` to `sum` or `result` or something more expressive? 2. We are adding a number to a boolean here, WDYT about:
```suggestion
        messagesNavButton._hasNotification = VPNAddonManager.reduce((addon, sum) => {
            if (addon.type === "message" && !addon.isRead) {
               sum++;
            }
            return sum;
           }, 0)
```
There is no change in behaviour, it's just less confusing and dependant or weird implicit JavaScript type coersions i.e. `true -> 1`.

Reply:
Also, I just noticed this was actually broken, because it needs to account for dismissed messages too. Would you liek to fix that in this PR or file a follow up to do that?

Reply:
that works because when a message is dismissed, it's disabled, so the counting will change.